gobblin-compaction/src/main/java/gobblin/compaction/event/CompactionSlaEvent.java

```diff
@@ -0,0 +1,129 @@
+package gobblin.compaction.event;
+
+import java.io.IOException;
+import java.util.Map;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.mapreduce.Counter;
+import org.apache.hadoop.mapreduce.Counters;
+import org.apache.hadoop.mapreduce.Job;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.collect.Maps;
+
+import gobblin.compaction.mapreduce.avro.AvroKeyDedupReducer;
+import gobblin.configuration.ConfigurationKeys;
+import gobblin.configuration.State;
+import gobblin.metrics.event.sla.SlaEvent;
+import gobblin.metrics.event.sla.SlaEventKeys;
+
+
+public class CompactionSlaEvent extends SlaEvent {
+
+  private static final Logger LOG = LoggerFactory.getLogger(CompactionSlaEvent.class);
+  private static final String EVENT_NAME = "CompactionCompleted";
+  private static final String DEFAULT = StringUtils.EMPTY;
+
+  private Job job;
```

Comment:
job - used to get record count
jobProps - used to get COMPACTION_DEST_DIR, COMPACTION_TOPIC, COMPACTION_JOB_DEST_PARTITION, COMPACTION_JOB_INPUT_DIR, COMPACTION_COMPLETE_FILE_NAME, COMPACTION_DEDUPLICATE
fs - used to get the modification time of the compaction file.

I see a few problems with passing all this data.
1. The list of arguments to be passed in the constructor is too long. It often leads to bugs and is unreadable
2. If this class needs more data from jobProps we have to change the method signature of the constructor.
3. To derive the data needed by this class we have to write some code. For example to get the record count we need to get all the counters from job and check if the record counter exists. I see this class as a place to collect that data.